src/UI/View.hs

```diff
@@ -24,34 +29,27 @@ type View = Fix ViewF
 type AView a = Cofree ViewF a
 
 
-measureString :: Num a => String -> Size a
-measureString s = Size (fromIntegral (length s) * fontW) lineH
-  where (fontW, fontH) = (5, 8)
-        lineH = fontH + 5
-
-measureStringForWidth :: Real a => a -> String -> Size a
-measureStringForWidth maxW s = Size maxW (height line * fromInteger (ceiling (toRational (length s) / (toRational maxW / toRational (width char)))))
-  where char = Size 5 8
-        line = char + Size 10 5
-
-layoutView :: Real a => View -> Layout a (Size a)
-layoutView = cata $ \ view -> case view of
-  Text s -> inset margins (resizeable (\ maxSize ->
-    pure (fromMaybe <$> maybe measureString measureStringForWidth (width maxSize) s <*> maxSize)))
-  Label s -> inset margins (pure (measureString s))
-  Scroll axis child -> inset margins (case axis of
-    Just Vertical -> resizeable (\ (Size _ maxH) -> do
-      Size w h <- child
-      pure (Size w (fromMaybe h maxH)))
-    Just Horizontal -> resizeable (\ (Size maxW _) -> do
-      Size w h <- child
-      pure (Size (fromMaybe w maxW) h))
-    _ -> resizeable (\ maxSize -> do
-      childSize <- child
-      pure (fromMaybe <$> childSize <*> maxSize)))
-  List children -> inset margins (stack (intersperse (offset spacing (pure (Size 0 0))) children))
+renderView :: Real a => View -> Rendering a ()
+renderView = cata $ \ view -> inset margins $ case view of
+  Text s -> resizeable (`text` s)
+  Label s -> text (pure Nothing) s
+  List children -> stack (intersperse (offset spacing (pure ())) children)
+  Scroll axis child -> resizeable (\ (Size maxW maxH) ->
+    measure child (\ (Size w h) ->
+      clip (case axis of
+        Just Horizontal -> Size w (fromMaybe h maxH)
+        Just Vertical -> Size (fromMaybe w maxW) h
+        Nothing -> fromMaybe <$> Size w h <*> Size maxW maxH) child))
```

Comment:
View rendering ended up even simpler than view layout, because drawing text is enough to imply its layout.

I’m particularly fond of the trick differentiating `Text` and `Label`—in `Text`, the drawn text is resized to fit. In `Label`, it isn’t.